examples/data-objects/dataStoreRace/src/index.tsx

```diff
@@ -0,0 +1,248 @@
+        const counterHandle = this.root.get<IFluidHandle<SharedCounter>>(counterKey);
+        this._counter = await counterHandle.get();
+        console.log(`counter ${this._counter?.value}`);
+    }
+
+    public render(div: HTMLElement) {
+        // Get our counter object that we set in initialize and pass it in to the view.
+        ReactDOM.render(
+            <CounterReactView counter={this.counter} />,
+            div,
+        );
+        return div;
+    }
+
+    private get counter() {
+        if (this._counter === undefined) {
+            throw new Error("SharedCounter not initialized");
+        }
+        return this._counter;
+    }
+
+    public merge(target: this) {
+        target.counter.increment(this.counter.value);
+}
+
+    public dispose() {
+        this.runtime.dispose();
+    }
+
+    public snapshot() {
+        return this.runtime.getAttachSnapshot();
+    }
+}
+
+// ----- REACT STUFF -----
+
+interface CounterProps {
+    counter: SharedCounter;
+}
+
+interface CounterState {
+    value: number;
+}
+
+class CounterReactView extends React.Component<CounterProps, CounterState> {
+    constructor(props: CounterProps) {
+        super(props);
+
+        this.state = {
+            value: this.props.counter.value,
+        };
+    }
+
+    componentDidMount() {
+        this.props.counter.on("incremented", (incrementValue: number, currentValue: number) => {
+            this.setState({ value: currentValue });
+        });
+    }
+
+    render() {
+        return (
+            <div>
+                <span className="clicker-value-class" id={`clicker-value-${Date.now().toString()}`}>
+                    {this.state.value}
+                </span>
+                <button onClick={() => { this.props.counter.increment(1); }}>+</button>
+            </div>
+        );
+    }
+}
+
+const MerableClickerInstantiationFactory:
+    IFluidDataStoreFactory &
+    {
+        createLocal: (containerRuntime: ContainerRuntime)
+             => Promise<MergeableClicker>,
+    } = {
+    type: "MergableClicker",
+    get IFluidDataStoreFactory() {return this;},
+    instantiateDataStore: async (context: IFluidDataStoreContext) => {
+        const facts: IChannelFactory[] = [
+            SharedMap.getFactory(),
+            SharedCounter.getFactory(),
+        ];
+        const reg: ISharedObjectRegistry =
+            new Map<string, IChannelFactory | undefined>(facts.map((ext) => [ext.type, ext]));
+
+        const dsRuntime = FluidDataStoreRuntime.load(
+            context,
+            reg);
+
+        // cannot await this as we get in infinite loop of realization due to handle resolution
+        //
+        const clickerP = MergeableClicker.load(dsRuntime);
+
+        dsRuntime.registerRequestHandler(async (req) => (await clickerP).request(req));
+
+        return dsRuntime;
+    },
+    createLocal: async (containerRuntime: ContainerRuntime) => {
+        const loader = Loader._create(
```

Comment:
create a detached contianer with a normal runtime factory to get a fully local, and fully isolated instance of the data store. We'de need to extend the runtime to support creating these directly, but again, this was simple.